cromwell-backend/src/main/scala/cromwell/backend/provider/local/LocalBackend.scala

```diff
@@ -0,0 +1,348 @@
+      val outputsExpressions = task.outputs.map(
+        output => output.name ->(output.wdlType, output.expression.evaluate(lookupFunction, expressionEval)))
+      processOutputResult(processReturnCode, outputsExpressions)
+    }
+  }
+
+  /**
+    * Process output evaluating expressions, checking for created files and converting WdlString to WdlSimpleFile if necessary.
+    * @param processReturnCode Return code from process.
+    * @param outputsExpressions Outputs.
+    * @return TaskStatus with final status of the task.
+    */
+  private def processOutputResult(processReturnCode: Int, outputsExpressions: Seq[(String, (WdlType, Try[WdlValue]))]): TaskStatus = {
+    if (outputsExpressions.filter(_._2._2.isFailure).size > 0) {
+      TaskStatus(Status.Failed, Some(FailureResult(new IllegalStateException("Failed to evaluate output expressions.",
+        outputsExpressions.filter(_._2._2.isFailure).head._2._2.failed.get), Some(processReturnCode), stderr.toString)))
+    } else {
+      try {
+        TaskStatus(Status.Succeeded, Some(SuccesfulResult(outputsExpressions.map(
+          output => output._1 -> resolveOutputValue(output._2)).toMap)))
+      } catch {
+        case ex: Exception => TaskStatus(Status.Failed, Some(FailureResult(ex, Some(processReturnCode), stderr.toString)))
+      }
+    }
+  }
+
+  /**
+    * 1) Remove all leading newline chars
+    * 2) Remove all trailing newline AND whitespace chars
+    * 3) Remove all *leading* whitespace that's common among every line in the input string
+    *
+    * For example, the input string:
+    *
+    * "
+    * first line
+    * second line
+    * third line
+    *
+    * "
+    *
+    * Would be normalized to:
+    *
+    * "first line
+    * second line
+    * third line"
+    *
+    * @param s String to process
+    * @return String which has common leading whitespace removed from each line
+    */
+  private def normalize(s: String): String = {
+    val trimmed = stripAll(s, "\r\n", "\r\n \t")
+    val parts = trimmed.split("[\\r\\n]+")
+    val indent = parts.map(leadingWhitespaceCount).min
+    parts.map(_.substring(indent)).mkString("\n")
+  }
+
+  private def leadingWhitespaceCount(s: String): Int = {
+    val Ws = Pattern.compile("[\\ \\t]+")
+    val matcher = Ws.matcher(s)
+    if (matcher.lookingAt) matcher.end else 0
+  }
+
+  private def stripAll(s: String, startChars: String, endChars: String): String = {
+    /* https://stackoverflow.com/questions/17995260/trimming-strings-in-scala */
+    @tailrec
+    def start(n: Int): String = {
+      if (n == s.length) ""
+      else if (startChars.indexOf(s.charAt(n)) < 0) end(n, s.length)
+      else start(1 + n)
+    }
+
+    @tailrec
+    def end(a: Int, n: Int): String = {
+      if (n <= a) s.substring(a, n)
+      else if (endChars.indexOf(s.charAt(n - 1)) < 0) s.substring(a, n)
+      else end(a, n - 1)
+    }
+
+    start(0)
+  }
```

Comment:
I'm confused about why `normalize()`, `stripAll()`, etc are in LocalBackend.  Shouldn't every command string have `normalize` called on it before being executed on *any* backend?  This is especially true for us, as we tend to do this in WDL a lot:

```
task test {
  command {
    python <<CODE
    print('hello')
    CODE
  }
}
```

And without the `normalize()` being called on this command, the leading whitespace would cause the python command to fail.